tests/avif/compare_avif_to_png.c

```diff
@@ -0,0 +1,93 @@
+/**
+ * File: compare_avif_to_png
+ *
+ * Thorough check for AVIF encoding and decoding.
+ * This test reqiures a set of PNG images that have been losslessly encoded to AVIFs.
+ * For each such image, we encode the PNG into an AVIF, with the GD format as an intermediary,
+ * then compare the resulting AVIF with the original PNG.
+ *
+ * We then do this process in reverse, encoding the AVIF into a PNG,
+ * and compare the resulting PNG with the original AVIF.
+ *
+ * We report any discrepancies in the images, or any other errors that may occur.
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include "gd.h"
+#include "gdtest.h"
+
+#define PATH "avif/"
+#define MAX_FILEPATH_LENGTH 200
+
+int main() {
+	FILE *fp;
+	gdImagePtr imFromPng = NULL, imFromAvif = NULL;
+	void *avifImDataPtr = NULL, *pngImDataPtr = NULL;
+	int size;
+	char filePath[MAX_FILEPATH_LENGTH], pngFilePath[MAX_FILEPATH_LENGTH], avifFilePath[MAX_FILEPATH_LENGTH];
+  char errMsg[MAX_FILEPATH_LENGTH + 100];
```

Comment:
the whitespace in this file is all funky

Reply:
I keep grepping for bad whitespace, and yet somehow I sometimes miss some 😭